corehq/apps/users/models.py

```diff
@@ -538,7 +539,10 @@ def add_domain_membership(self, domain, timezone=None, **kwargs):
                 return
 
         domain_obj = Domain.get_by_name(domain, strict=True)
-        if not domain_obj:
+        _soft_assert = soft_assert(notify_admins=True)
+        if not _soft_assert(domain_obj,
+                            "Domain membership added before domain created",
+                            {'domain': domain}):
```

Comment:
might be good to also leave a comment, e.g. "if you haven't seen any of these by Feb 2016 you should delete this code.